src/force/nep3.cu

```diff
@@ -457,7 +463,12 @@ static __global__ void find_partial_force_angular(
           gn12 += fn12[k] * annmb.c[c_index];
           gnp12 += fnp12[k] * annmb.c[c_index];
         }
-        accumulate_f12_nep3(n, paramb.n_max_angular + 1, d12, r12, gn12, gnp12, Fp, sum_fxyz, f12);
+        if (paramb.num_L > paramb.L_max) {
+          accumulate_f12_with_4body(
+            n, paramb.n_max_angular + 1, d12, r12, gn12, gnp12, Fp, sum_fxyz, f12);
+        } else {
+          accumulate_f12(n, paramb.n_max_angular + 1, d12, r12, gn12, gnp12, Fp, sum_fxyz, f12);
```

Comment:
And these are the forces on a single atom?

Reply:
This is the "partial force"  dU_i/d \vec{r}_ij I defined in the manuscript  for atom i coming from atom j only. There will be summation over j in an outer loop (hence the name "accumulate")

Reply:
And it is only for the angular part, that is, the last two terns on the right hand side of Eq. (18) of the manuscript.

Reply:
Alright, makes sense :)